dialects/core/be/ll/ll.cpp

```diff
@@ -498,24 +493,14 @@ std::string Emitter::emit_bb(BB& bb, const Def* def) {
         if (lit->type()->isa<Nat>()) {
             return std::to_string(lit->get<nat_t>());
         } else if (auto size = Idx::size(lit->type())) {
-            if (size->isa<Top>()) return std::to_string(lit->get<nat_t>());
-            if (auto mod = size2bitwidth(as_lit(size))) {
-                switch (*mod) {
-                    // clang-format off
-                    case  0: return {};
-                    case  1: return std::to_string(lit->get< u1>());
-                    case  2:
-                    case  4:
-                    case  8: return std::to_string(lit->get< u8>());
-                    case 16: return std::to_string(lit->get<u16>());
-                    case 32: return std::to_string(lit->get<u32>());
-                    case 64: return std::to_string(lit->get<u64>());
-                    // clang-format on
-                    default: return std::to_string(lit->get<u64>());
-                }
-            } else {
-                return std::to_string(lit->get<u64>());
+            if (auto s = isa_lit(size)) {
```

Comment:
Code duplication with other points where s is converted to bits.

Reply:
This is hard to factor out due to the template arguments involved.

Reply:
Alternatively, one could use a `typename std::conditional` construct for `uint8_t -> type` and use this.
I think this would be better as it isolated the different logic and moves the conversion with (semi-)magic numbers to a single point.